promql/engine_test.go

```diff
@@ -565,15 +567,14 @@ func TestSelectHintsSetCorrectly(t *testing.T) {
 				query Query
 				err   error
 			)
-			ctx := context.Background()
 			if tc.end == 0 {
-				query, err = engine.NewInstantQuery(ctx, hintsRecorder, nil, tc.query, timestamp.Time(tc.start))
+				query, err = engine.NewInstantQuery(ctxbg, hintsRecorder, nil, tc.query, timestamp.Time(tc.start))
```

Comment:
Let's leave context.Background line, it's just one extra line, it's fine to keep it.

The "bad" consequence of optimizing one line is that this context looks "special", so reader would be double checking where this context comes from does it has something special (e.g. values or timeouts)--whereas it's just simple context.

Reply:
:heavy_check_mark: